pkg/vsphere/tasks/waiter.go

```diff
@@ -91,15 +91,19 @@ func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (
 
 func isTaskInProgress(err error) bool {
 	if soap.IsSoapFault(err) {
-		if _, ok := soap.ToSoapFault(err).VimFault().(types.TaskInProgress); ok {
+		f := soap.ToSoapFault(err)
+		if _, ok := f.VimFault().(types.TaskInProgress); ok {
 			return true
 		}
+		log.Debugf("unexpected soap fault on task retry : %s", spew.Sdump(f))
```

Comment:
What does this output look like?  I think at this point, where the fault has been unwrapped from Soap/Vim, just using '%#v' with 'f' should provide enough details here.  That's what we're doing elsewhere.  Not sure its worth introducing go-spew here, but if it is we ought to use it across the board - and maybe hidden behind a helper so it'll be simple to adjust, swap in a different pretty printer if needed, etc.